code/modules/mob/living/carbon/human/human_organs.dm

```diff
@@ -173,17 +173,23 @@
 		O.set_dna(dna)
 
 /mob/living/carbon/human/is_asystole()
-	if(isSynthetic())
-		var/obj/item/organ/internal/cell/C = internal_organs_by_name[BP_CELL]
-		if(istype(C))
-			if(!C.is_usable())
-				return TRUE
-	else if(should_have_organ(BP_HEART))
-		var/obj/item/organ/internal/heart/heart = internal_organs_by_name[BP_HEART]
+	if(should_have_process(OP_HEART))
+		var/obj/item/organ/internal/heart/heart = random_organ_by_process(OP_HEART)
 		if(!istype(heart) || !heart.is_working())
 			return TRUE
 	return FALSE
 
+/mob/living/carbon/human/proc/organ_list_by_process(organ_process)
+	RETURN_TYPE(/list)
+	. = list()
+	for(var/organ in internal_organs_by_efficiency[organ_process])
+		. += organ
+
+/mob/living/carbon/human/proc/random_organ_by_process(organ_process)
+	if(organ_list_by_process(organ_process).len)
+		return pick(organ_list_by_process(organ_process))
```

Comment:
same for this proc. I do not know if it is because it is designed in this way but since we do not have range of vision for each eye it would be best to simply use the average of the efficiency of all the eyes.

Reply:
Same as above.